www/js/app.js

```diff
@@ -1051,15 +1072,23 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
                     $("#searchingArticles").show();
                 };
             };
+
+            if(! isDirEntryExpectedToBeDisplayed(dirEntry)){
+                return;
+            } 
+
             // We put the ZIM filename as a prefix in the URL, so that browser caches are separate for each ZIM file
             iframeArticleContent.src = "../" + selectedArchive._file._files[0].name + "/" + dirEntry.namespace + "/" + encodedUrl;
         } else {
             // In jQuery mode, we read the article content in the backend and manually insert it in the iframe
             if (dirEntry.isRedirect()) {
-                selectedArchive.resolveRedirect(dirEntry, readArticle);
+                // Update expectedArticleURLToBeDisplayed before redirect then call readArticle.
+                selectedArchive.resolveRedirect(dirEntry, function (resolvedDirEntry) {
+                    expectedArticleURLToBeDisplayed = resolvedDirEntry.namespace + "/" + resolvedDirEntry.url;
+                    readArticle(resolvedDirEntry)
+                });
             } else {
                 // Line below was inserted to prevent the spinner being hidden, possibly by an async function, when pressing the Random button in quick succession
-                // TODO: Investigate whether it is really an async issue or whether there is a rogue .hide() statement in the chain
```

Comment:
Why did you remove this comment?
It was a comment from @Jaifroid, please leave it unless he asked you to remove it